src/cursor/cur_std.c

```diff
@@ -1196,59 +1179,48 @@ __wt_cursor_largest_key(WT_CURSOR *cursor)
 int
 __wt_cursor_bound(WT_CURSOR *cursor, const char *config)
 {
+    WT_CONFIG_ITEM cval;
     WT_CURSOR_BTREE *cbt;
+    WT_DECL_CONF(WT_CURSOR, bound, conf);
     WT_DECL_RET;
     WT_ITEM key;
     WT_SESSION_IMPL *session;
     int exact;
-    char *cfg;
-    bool inclusive, have_action;
+    bool inclusive;
 
-    cfg = NULL;
     cbt = (WT_CURSOR_BTREE *)cursor;
     exact = 0;
-
-    /*
-     * Our API defines "inclusive" as true by default, it also defines "set" as the default action.
-     * This means we can't expect the user to have provided those configurations via the config
-     * string. Normally WiredTiger merges the user provided configuration with the default
-     * configuration, for performance reasons we are skipping this step. As such we need to handle
-     * the cases where the user did and didn't provide the config making up the difference for the
-     * defaults.
-     *
-     * These two booleans provide a mechanism to handle the user not providing the configuration and
-     * still being able to parse it.
-     */
     inclusive = true;
-    have_action = false;
 
     CURSOR_API_CALL(cursor, session, ret, bound, NULL);
+    WT_ERR(__wt_conf_compile_api_call(session, WT_CONFIG_REF(session, WT_CURSOR_bound),
+      WT_CONFIG_ENTRY_WT_CURSOR_bound, config, &_conf, sizeof(_conf), &conf));
```

Comment:
I think this should be the simpler:
```
  API_CONF(session, WT_CURSOR, bound, cfg, conf);
```

Reply:
I deliberately skipped this as it depends on calling `CURSOR_API_CONF` which is about a 20% perf hit. The reason it depends on that is it de-references the cfg array: `cfg[1]` which is only built if you go through the more expensive API call. Is it okay if we leave it out? It may also point to further config API improvements.

Reply:
Ah, interesting.  Yes, leave it out.  I'd like to explore this in another ticket.

Reply:
WT-12784